src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs

```diff
@@ -886,7 +904,24 @@ private void DeleteButtonClick()
 
 		partial void OnDeleteButtonClickPartial();
 
-		internal void OnSelectionChanged() => SelectionChanged?.Invoke(this, new RoutedEventArgs(this));
+		internal void OnSelectionChanged()
+		{
+			if(!string.IsNullOrEmpty(Text))
+			{
+				if (!Text.TryGetStringByRange(SelectionStart, SelectionLength, out _))					
+				{
+					this.ResetCarret();
+				}
+				else
+				{
+					SelectionChanged?.Invoke(this, new RoutedEventArgs(this));
+				}
+			}
+			else
+			{
+				SelectionChanged?.Invoke(this, new RoutedEventArgs(this));
+			}
```

Comment:
Avoid processing the _negative_ clause first...  it can easily confuse the reading of the code.

```suggestion
			if(Text is { Length: > 0 } text || text.TryGetStringByRange(SelectionStart, SelectionLength, out _))
			{
				SelectionChanged?.Invoke(this, new RoutedEventArgs(this));
			}
			else
			{
				ResetCarret();
			}
```

Reply:
The suggestion was commited.